public/admin/permalink.haml

```diff
@@ -2,52 +2,52 @@
 - if flash[:error]
   #error
     %p= flash[:error]
-%form{'action' => url('/admin/permalink'), 'method' => 'post', 'data-method' => 'put'}
-  %input{'name' => '_method', 'type' => 'hidden', 'value' => 'put'}
+%form{ action: url('/admin/permalink'), method: :post, 'data-method': :put }
+  %input{ name: :_method, type: :hidden, value: :put }
   %p
     = t('permalink.description')
   %p
-    %input{'name' => 'enable', 'type' => 'hidden', 'value' => 0}
-    %input#permalink_enable{'name' => 'enable', 'type' => 'checkbox', 'value' => '1', 'checked' => @enabled||nil}
-    %label{'for' => 'permalink_enable'}= t('permalink.enable')
+    %input{ name: :enable, type: :hidden, value: 0 }
+    %input#permalink_enable{ name: :enable, type: :checkbox, value: '1', checked: @enabled || nil }
+    %label{ for: :permalink_enable }= t('permalink.enable')
   %p
-    %label{'for' => 'permalink_format'}= t('permalink.format')
+    %label{ for: :permalink_format }= t('permalink.format')
     %br/
-    %input#permalink_format{'name' => 'format', 'type' => 'text', 'value' => flash[:permalink_format] || @format}
+    %input#permalink_format{ name: :format, type: :text, value: flash[:permalink_format] || @format }
```

Comment:
I made this ignorable by setting `IdNames` linter disabled in `.haml-lint_todo.yml`. Because it may have unpredictable side effects on JavaScript and CSS.